civo/instances/resource_instance.go

```diff
@@ -468,6 +469,56 @@ func resourceInstanceUpdate(ctx context.Context, d *schema.ResourceData, m inter
 		}
 	}
 
+	// If reserved_ipv4 has changed, update the instance with the new reserved IP
+	if d.HasChange("reserved_ipv4") {
+		oldReservedIP, newReservedIP := d.GetChange("reserved_ipv4")
+		instance, err := apiClient.GetInstance(d.Id())
+		if err != nil {
+			// Check if the instance no longer exists.
+			return diag.Errorf("[ERR] instance %s not found", d.Id())
+		}
+
+		// Unassign the old reserved IP if it exists
+		if oldReservedIP != "" {
+			ip, err := apiClient.FindIP(oldReservedIP.(string))
+			if err != nil {
+				if errors.Is(err, civogo.ZeroMatchesError) {
+					return diag.Errorf("sorry there is no %s IP in your account", oldReservedIP)
```

Comment:
I would suggest to rework and reword this. Something like "Sorry, there is no IP that matches %s in in this network" ?

I suggest "network" rather than account as these are network-specific, right?